Example/ViewController.swift

```diff
@@ -258,11 +258,14 @@ class ViewController: UIViewController {
     func presentAndRemoveMapview(_ navigationViewController: NavigationViewController) {
         let delegate = UIApplication.shared.delegate as? AppDelegate
         
-        if #available(iOS 12.0, *), let service = navigationViewController.navigationService, let location = service.router.location {
-            delegate?.carPlayManager.beginNavigationWithCarPlay(using: location.coordinate, navigationService: service)
-        }
-        
         present(navigationViewController, animated: true) {
+            if #available(iOS 12.0, *),
+                let service = navigationViewController.navigationService,
+                let location = service.router.location {
+                delegate?.carPlayManager.beginNavigationWithCarPlay(using: location.coordinate,
```

Comment:
The documentation for this method should have a `precondition` about the NavigationViewController already having been presented fully.